skimage/measure/fit.py

```diff
@@ -251,18 +316,38 @@ class CircleModel(BaseModel):
     --------
     >>> t = np.linspace(0, 2 * np.pi, 25)
     >>> xy = CircleModel().predict_xy(t, params=(2, 3, 4))
-    >>> model = CircleModel()
-    >>> model.estimate(xy)
-    True
+    >>> model = CircleModel.from_estimate(xy)
     >>> tuple(np.round(model.params, 5))
     (2.0, 3.0, 4.0)
     >>> res = model.residuals(xy)
     >>> np.abs(np.round(res, 9))
     array([0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
            0., 0., 0., 0., 0., 0., 0., 0.])
+
+    The estimation can fail when — for example — all the input or output
+    points are the same.  If this happens, you will get a circl model for which
+    ``bool(model)`` is ``False``:
+
+    >>> # bool on a successful model (as above) gives True:
+    >>> bool(model)
```

Comment:
I don't think we really want to encourage that pattern of explicitly converting to bool. Of course, it's correct, just that we want the spelling to be the following:

```
>>> if not model:
...     print("Estimation failed")
```

Reply:
+1 – the `if not model` version should clearer to many users not aware of `__bool__`.